build_defs/internal_shell.bzl

```diff
@@ -0,0 +1,85 @@
+def inline_sh_binary(
```

Comment:
Possibly?  This took a lot of iterations and tweaking it get it to work right and be usable/debuggable.  IIRC the main issue with using a genrule alone is that it can't be run as a single binary in *other* commands (i.e. $(location :foo) resolves to multiple files).

Since this is an internal util that's only meant to be a temporary migration mechanism, I'm inclined to just leave it as is.  Many of the followup tasks boil down to simply "remove this inline_sh_binary rule", so they serve as a pretty good marker of what's left to bazelify.